pulsar-client-cpp/CMakeLists.txt

```diff
@@ -23,13 +23,20 @@ project (pulsar-cpp)
 option(LINK_STATIC "Link against static libraries" OFF)
 MESSAGE(STATUS "LINK_STATIC:  " ${LINK_STATIC})
 
+IF (CMAKE_BUILD_TYPE STREQUAL "")
```

Comment:
IF (CMAKE_BUILD_TYPE STREQUAL "")
    MESSAGE(STATUS "CMAKE_BUILD_TYPE is set to default (Release)")
else 
    MESSAGE(STATUS "CMAKE_BUILD_TYPE:  " ${CMAKE_BUILD_TYPE})
ENDIF (CMAKE_BUILD_TYPE STREQUAL "")



`cmake -DCMAKE_BUILD_TYPE= RelWithDebInfo` when we run it in docker 

The reason is that the default value is Release.

------ Rest is ok